From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Thu, 17 Jul 2014 22:59:47 +0200
Subject: [199/249] staging: rtl8723au: Use rtl8723au_set_media_status() to
 replace _InitNetworkType()
Origin: https://git.kernel.org/linus/df72ac93ee22daf1595e93b45de9e6804af0afde

Reduce yet another duplicate implementation of set_media_status()

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/hal/usb_halinit.c       | 13 +------------
 drivers/staging/rtl8723au/include/rtl8723a_spec.h |  8 --------
 2 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/staging/rtl8723au/hal/usb_halinit.c b/drivers/staging/rtl8723au/hal/usb_halinit.c
index 8a7e644..d8f9f27 100644
--- a/drivers/staging/rtl8723au/hal/usb_halinit.c
+++ b/drivers/staging/rtl8723au/hal/usb_halinit.c
@@ -338,17 +338,6 @@ static void _InitQueuePriority(struct rtw_adapter *Adapter)
 	_InitNormalChipQueuePriority(Adapter);
 }
 
-static void _InitNetworkType(struct rtw_adapter *Adapter)
-{
-	u32 value32;
-
-	value32 = rtl8723au_read32(Adapter, REG_CR);
-
-	/*  TODO: use the other function to set network type */
-	value32 = (value32 & ~MASK_NETTYPE) | _NETTYPE(NT_LINK_AP);
-	rtl8723au_write32(Adapter, REG_CR, value32);
-}
-
 static void _InitTransferPageSize(struct rtw_adapter *Adapter)
 {
 	/*  Tx page size is always 128. */
@@ -756,7 +745,7 @@ static int rtl8723au_hal_init(struct rtw_adapter *Adapter)
 
 	_InitInterrupt(Adapter);
 	hw_var_set_macaddr(Adapter, Adapter->eeprompriv.mac_addr);
-	_InitNetworkType(Adapter);/* set msr */
+	rtl8723a_set_media_status(Adapter, MSR_INFRA);
 	_InitWMACSetting(Adapter);
 	_InitAdaptiveCtrl(Adapter);
 	_InitEDCA(Adapter);
diff --git a/drivers/staging/rtl8723au/include/rtl8723a_spec.h b/drivers/staging/rtl8723au/include/rtl8723a_spec.h
index b5d7123..8d19b28 100644
--- a/drivers/staging/rtl8723au/include/rtl8723a_spec.h
+++ b/drivers/staging/rtl8723au/include/rtl8723a_spec.h
@@ -1323,14 +1323,6 @@ Current IOREG MAP
 #define ENSWBCN				BIT(8)
 #define ENSEC				BIT(9)
 
-/*  Network type */
-#define _NETTYPE(x)			(((x) & 0x3) << 16)
-#define MASK_NETTYPE			0x30000
-#define NT_NO_LINK			0x0
-#define NT_LINK_AD_HOC			0x1
-#define NT_LINK_AP			0x2
-#define NT_AS_AP			0x3
-
 #define _LBMODE(x)			(((x) & 0xF) << 24)
 #define MASK_LBMODE			0xF000000
 #define LOOPBACK_NORMAL			0x0
